components/org.wso2.carbon.identity.login.resolver/org.wso2.carbon.identity.login.resolver.regex/src/main/java/org/wso2/carbon/identity/login/resolver/regex/RegexLoginResolver.java

```diff
@@ -0,0 +1,205 @@
+
+/**
+ * This class defines the default implementation of the LoginResolver. With the help of this login resolver, users will
+ * get resolved using a claim regex pattern defined in the management console.
+ */
+public class RegexLoginResolver implements LoginResolver {
+
+    private static final Log log = LogFactory.getLog(RegexLoginResolver.class);
+
+    /**
+     * Resolves a user from the given login identifier and then returns the resolved claim URI and the user details if
+     * a matching user exists.
+     *
+     * @param loginIdentifier   The user login identifier input which is provided by the user during login.
+     * @param allowedAttributes The list of claim URIs which are allowed by admin for resolving the user.
+     * @param tenantDomain      The tenant domain of the user.
+     * @return A ResolvedUserResult object with the user details and the user resolved claim details.
+     */
+    @Override
+    public ResolvedUserResult resolveUser(String loginIdentifier, List<String> allowedAttributes, String tenantDomain) {
+
+        ResolvedUserResult resolvedUserResult = new ResolvedUserResult(ResolvedUserResult.UserResolvedStatus.FAIL);
+        try {
+            if (allowedAttributes == null) {
+                return resolvedUserResult;
+            }
+            UserRealm userRealm = UserResolverUtil.getUserRealm(tenantDomain);
+            UniqueIDUserStoreManager userStoreManager = UserResolverUtil.getUserStoreManager(tenantDomain);
+            ClaimManager claimManager = userRealm.getClaimManager();
+            for (String claimURI : allowedAttributes) {
+                Claim claim = claimManager.getClaim(claimURI);
+                if (claim == null) {
+                    continue;
+                }
+                String regex = claim.getRegEx();
+                if (StringUtils.isBlank(regex)) {
+                    continue;
+                }
+                Pattern pattern = Pattern.compile(regex);
+                String domainSeparateAttribute = UserCoreUtil.removeDomainFromName(loginIdentifier);
+                if (pattern.matcher(domainSeparateAttribute).matches()) {
+                    setResolvedUserResult(userStoreManager, claimURI, loginIdentifier, resolvedUserResult, claim);
+                    break;
+                }
+            }
+            /*
+            resolve user if allowed attributes has only username claim,
+            but username claim has no configured regex pattern.
+             */
+            if (allowedAttributes.size() == 1 &&
+                    allowedAttributes.contains(UserCoreClaimConstants.USERNAME_CLAIM_URI)) {
+                setResolvedUserResult(userStoreManager, UserCoreClaimConstants.USERNAME_CLAIM_URI, loginIdentifier,
+                        resolvedUserResult, claimManager.getClaim(UserCoreClaimConstants.USERNAME_CLAIM_URI));
+            }
+        } catch (UserStoreException e) {
+            log.error("An error occurred while resolving user.", e);
```

Comment:
BTW here we just log the error and allow to continue the process. should it be stopped here when we get this error ? WDYT ?

Reply:
From my understanding when this exception is thrown a failure result will be sent out since the `resolvedUserResult` is by default set to the status `ResolvedUserResult.UserResolvedStatus.FAIL`.